src/components/templates/PosterPage/components/PosterPageSettingsControl/PosterPageSettingsControl.tsx

```diff
@@ -1,45 +1,70 @@
-import React from "react";
+import React, { useCallback } from "react";
 import { OverlayTrigger, Popover } from "react-bootstrap";
 import { faCog } from "@fortawesome/free-solid-svg-icons";
 
+import { setVenueLiveStatus } from "api/venue";
+
 import { Checkbox } from "components/atoms/Checkbox";
 
 import { PosterPageControl } from "../PosterPageControl";
 
 import "./PosterPageSettingsControl.scss";
 
 export interface PosterPageSettingsControlProps {
-  setVenueLiveOff: () => void;
-  setVenueLiveOn: () => void;
+  venueId: string;
   isPosterLive?: boolean;
 }
 
 export const PosterPageSettingsControl: React.FC<PosterPageSettingsControlProps> = ({
-  isPosterLive,
-  setVenueLiveOff,
-  setVenueLiveOn,
+  isPosterLive = false,
+  venueId,
 }) => (
   <OverlayTrigger
     trigger="click"
     placement="bottom-end"
     overlay={
-      <Popover
-        id="poster-page-settings-popover"
-        className="PosterPageSettingsControl__popover"
-      >
-        <Popover.Content className="PosterPageSettingsControl__popover-content">
-          <Checkbox
-            type="checkbox"
-            checked={!!isPosterLive}
-            onChange={isPosterLive ? setVenueLiveOff : setVenueLiveOn}
-            title={isPosterLive ? "Poster is live" : "Make poster live"}
-            toggler
-          />
-        </Popover.Content>
-      </Popover>
+      <PosterPageSettingsControlOverlay
+        isPosterLive={isPosterLive}
+        venueId={venueId}
+      />
     }
     rootClose={true}
   >
     <PosterPageControl label="Settings" icon={faCog} />
   </OverlayTrigger>
 );
+
+export interface PosterPageSettingsControlOverlayProps {
+  isPosterLive?: boolean;
+  venueId: string;
```

Comment:
Order + `isPosterLive` will always be a `boolean` (not `undefined`) here since it's defaulted in the parent component.

```suggestion
  venueId: string;
  isPosterLive: boolean;
```

Reply:
done in https://github.com/sparkletown/sparkle/pull/1299/commits/4100b1fbb55e67940252f7ebc610231c4e7bb993

Reply:
> done in 4100b1f

I'm not sure that is correct since you reverted the changes that extracted this into 2 components?

My comments above were about the inner component not needing the optional/default because the outer component was optional/defaulting it. So for the inner component it would always be set.

But now there is only 1 component, it's effectively the outer component.

Basically, if you're going to keep this as just the single component, then you may as well revert 4100b1f as it was better/correct before it.